tests/preprocessing/test_positionfixes.py

```diff
@@ -1,116 +1,225 @@
-        pfs_for, spts_for = _generate_staypoints_original(pfs_ori, 
-                                                          method='sliding', 
-                                                          dist_threshold=25, 
+        pfs_for, spts_for = _generate_staypoints_original(pfs_ori,
+                                                          method='sliding',
+                                                          dist_threshold=25,
                                                           time_threshold=300)
-        
+
         pd.testing.assert_frame_equal(spts_groupby, spts_for, check_dtype=False)
         pd.testing.assert_frame_equal(pfs_groupby, pfs_for, check_dtype=False)
-        
+
     def test_generate_staypoints_groupby_dbscan(self):
         """Test the 'dbscan' result obtained using user_id for loop (previous) with groupby.apply (current)."""
         pfs_file = os.path.join('tests', 'data', 'positionfixes.csv')
         pfs_ori = ti.read_positionfixes_csv(pfs_file, sep=';', tz='utc', index_col='id')
-        
+
         # stps detection using groupby
         pfs_groupby, spts_groupby = pfs_ori.as_positionfixes.generate_staypoints(method='dbscan')
         # stps detection using for loop
         pfs_for, spts_for = _generate_staypoints_original(pfs_ori, method='dbscan')
-        
+
         pd.testing.assert_frame_equal(spts_groupby, spts_for, check_dtype=False)
         pd.testing.assert_frame_equal(pfs_groupby, pfs_for, check_dtype=False)
-        
+
 
 class TestGenerate_triplegs():
-    def test_generate_triplegs_global(self):
-        # generate triplegs from raw-data
-        pfs = ti.io.dataset_reader.read_geolife(os.path.join('tests', 'data', 'geolife'))
-        pfs, spts = pfs.as_positionfixes.generate_staypoints(method='sliding', 
-                                                             dist_threshold=25, 
-                                                             time_threshold=5 * 60)
-        pfs, tpls = pfs.as_positionfixes.generate_triplegs(spts)
+    def test_generate_triplegs_case1(self, geolife_pfs_spts_short, testdata_tpls_geolife):
+        """Test tripleg generation using the 'between_staypoints' method for case 1.
+        case 1: triplegs from positionfixes with column 'staypoint_id' and staypoints."""
 
-        # load pregenerated test-triplegs
-        tpls_file = os.path.join('tests', 'data', 'geolife', 'geolife_triplegs_short.csv')
-        tpls_test = ti.read_triplegs_csv(tpls_file, tz='utc')
+        pfs, spts = geolife_pfs_spts_short
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(spts, method='between_staypoints')
 
-        assert len(tpls) > 0
-        assert len(tpls) == len(tpls)
+        assert (tpls.geometry.geom_almost_equals(testdata_tpls_geolife.geometry)).all()
+        pd.testing.assert_frame_equal(tpls.drop('geom', axis=1), testdata_tpls_geolife.drop('geom', axis=1),
+                                      check_like=True)
```

Comment:
Line 127-129 doing the same thing as function `assert_geodataframe_equal`?
Also Line 139-141, Line 150-152, Line 164-166, Line 181-183, Line 194-196

Reply:
Yes ;-) I wrote the `assert_geodataframe_equal` later when it annoyed me but I never used it for the tests that where already written. I fixed that now.